### PR TITLE
DOC: add pivot_table note to dropna doc

### LINKED ISSUE #47447: BUG: pd.pivot_table margins don't include rows, which have a NaN within any column, for all columns
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

df = pd.DataFrame(dict(index1=["a", "b", "c", "b", "a"], index2=["a", "b", "c", "b", "a"], value1=[10, 11, 12, 13, 14], value2=[15, 16, 17, 18, 19], valueNaN=[np.nan, 100, np.nan, 100, np.nan]))
print(df)
piv = pd.pivot_table(df, values=["value1", "value2", "valueNaN"], index=["index1", "index2"], aggfunc='sum', margins=True)
print(piv)
```

### Issue Description

df:
```
  index1 index2  value1  value2  valueNaN
0      a      a      10      15       NaN
1      b      b      11      16     100.0
2      c      c      12      17       NaN
3      b      b      13      18     100.0
4      a      a      14      19       NaN
```

Creating a pivot table with margins on multiple value columns within one function call, leads to the effect that the margins do not include rows, which have a NaN at **any** column, for **all** columns.
The above example leads to margins 24 and 34 for columns value1 and value 2
```
               value1  value2  valueNaN
index1 index2                          
a      a           24      34       0.0
b      b           24      34     200.0
c      c           12      17       0.0
All                24      34     200.0
```

### Expected Behavior

Expectation is that the margins include all values of columns value1 and value2, i.e. 24+24+12=60 and 34+34+17=85, instead of merely displaying the figures from index ('b', 'b'), i.e. 24 and 34, just because the other rows include NaN within the column valueNaN.
Expected results occur, when filling the NAs before creating the pivot:
`fixed_df = df.fillna(value=0)`
```
  index1 index2  value1  value2  valueNaN
0      a      a      10      15       0.0
1      b      b      11      16     100.0
2      c      c      12      17       0.0
3      b      b      13      18     100.0
4      a      a      14      19       0.0
```
`fixed_piv = pd.pivot_table(fixed_df, values=["value1", "value2", "valueNaN"], index=["index1", "index2"], aggfunc='sum', margins=True)`
```
               value1  value2  valueNaN
index1 index2                          
a      a           24      34       0.0
b      b           24      34     200.0
c      c           12      17       0.0
All                60      85     200.0
```

However, the current behavior, to not include rows with a NaN at all should not be default - or at least an appropriate information / warning should be provided for the pivot_table functionality.

### Installed Versions

<details>

pd.show_versions()
INSTALLED VERSIONS
------------------
commit           : 4bfe3d07b4858144c219b9346329027024102ab6
python           : 3.9.7.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19043
machine          : AMD64
processor        : AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252
pandas           : 1.4.2
numpy            : 1.22.3
pytz             : 2022.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 58.0.4
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : 1.3.4
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : None
matplotlib       : 3.5.1
numba            : None
numexpr          : 2.8.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.3
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by Mehgarg**
- [ ] closes #47447 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks for the pull request @Mehgarg 

The pandas documentation is generated automatically from markdown files and docstring (documentation in the code). We never edit the final html files directly.

I think the changes you want to implement should be done to this docstring instead: https://github.com/pandas-dev/pandas/blob/main/pandas/core/frame.py#L6427

Can you revert this html file you're creating and edit the docstring please? Let me know if you need help

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6455,7 +6455,7 @@ def dropna(
         subset : column label or sequence of labels, optional
             Labels along other axis to consider, e.g. if you are dropping rows
             these would be a list of columns to include.
-        inplace : bool, default False
+        inplace : bool, default False
```
Comment (datapythonista) [inline pandas/core/frame.py:6458]: Can you revert this?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6471,6 +6471,11 @@ def dropna(
         Series.dropna : Drop missing values.
         Index.dropna : Drop missing indices.
 
+        Notes
+        --------
+        Setting dropna to true will elimate rows containing any missing data 
+        from the pivot_table.
```
Comment (datapythonista) [inline pandas/core/frame.py:6477]: This doesn't seem to make a lot of sense. This method has nothing to do with pivot_table, and dropna is not a parameter, can't be set to anything. Maybe you may want to close this PR, try to understand better what you want to do, and start again?

File: doc/build/html/reference/api/pandas.DataFrame.dropna.html [hunk 3]
```diff
@@ -0,0 +1,415 @@
+
```
Comment (datapythonista) [inline doc/build/html/reference/api/pandas.DataFrame.dropna.html:1]: You should revert the changes to this whole file (delete it).

**Review by datapythonista — CHANGES_REQUESTED** [review]

**Comment by Mehgarg** [comment]
Is the failed check expected behavior or do I need to fix something?

File: pandas/core/frame.py [hunk 4]
```diff
@@ -8564,7 +8564,8 @@ def pivot(self, index=None, columns=None, values=None) -> DataFrame:
         margins : bool, default False
             Add all row / columns (e.g. for subtotal / grand totals).
         dropna : bool, default True
-            Do not include columns whose entries are all NaN.
+            Do not include columns whose entries are all NaN. If true:
+            exclude rows containing missing data from aggregated margins.
```
Comment (datapythonista) [inline pandas/core/frame.py:8568]: I think this is making things more confusing. I find just having the original description clearer. What is the problem you're trying to address here?

Comment (Mehgarg) [reply pandas/core/frame.py:8568]: I want to clarify for pviot_table, if any column within a row contains NaN, the entire row won't be included when aggregating margins. Currently, users expect values within the row not NaN to be aggregated.

Comment (datapythonista) [reply pandas/core/frame.py:8568]: I see. I don't think the description you added is veey clear, at least it wasn't for me. Something like `When this value is True, rows with a NaN value in any column will be discarded before computing margins.` maybe? Or maybe you can find something better.

File: pandas/core/frame.py [hunk 5]
```diff
@@ -10046,6 +10047,7 @@ def merge(
         copy: bool = True,
         indicator: bool = False,
         validate: str | None = None,
+
```
Comment (datapythonista) [inline pandas/core/frame.py:10050]: Can you revert this please? Better to avoid unrelated changes

File: pandas/core/frame.py [hunk 6]
```diff
@@ -10045,7 +10047,7 @@ def merge(
         suffixes: Suffixes = ("_x", "_y"),
         copy: bool = True,
         indicator: bool = False,
-        validate: str | None = None,
+        validate: str | None = None,
```
Comment (datapythonista) [inline pandas/core/frame.py:10050]: Can you please revert this?

File: pandas/core/frame.py [hunk 7]
```diff
@@ -8564,7 +8564,9 @@ def pivot(self, index=None, columns=None, values=None) -> DataFrame:
         margins : bool, default False
             Add all row / columns (e.g. for subtotal / grand totals).
         dropna : bool, default True
-            Do not include columns whose entries are all NaN.
+            Do not include columns whose entries are all NaN. If true:
```
Comment (datapythonista) [inline pandas/core/frame.py:8567]: True refers to Python True value, we usually use it with capital T. Also, please avoid the colon, as it's inconsistent with the rest of the docs, and personally I think make things harder to read in this case.

Other than those small formatting things, I think your description is clear now, thanks for thr update.

Comment (Mehgarg) [reply pandas/core/frame.py]: made the changes, thanks for the help.

File: pandas/core/frame.py (hunk 6 above)
Comment (datapythonista) [reply pandas/core/frame.py:10050]: This is pending and making the CI red

Comment (Mehgarg) [reply pandas/core/frame.py]: reverted

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @Mehgarg

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Mehgarg
